packages/ai-sdk/src/activities.ts

```diff
@@ -0,0 +1,103 @@
+import type {
+  LanguageModelV2CallOptions,
+  LanguageModelV2CallWarning,
+  LanguageModelV2Content,
+  LanguageModelV2FinishReason,
+  LanguageModelV2ResponseMetadata,
+  LanguageModelV2Usage,
+  ProviderV2,
+  SharedV2Headers,
+  SharedV2ProviderMetadata,
+} from '@ai-sdk/provider';
+import type { ToolCallOptions } from 'ai';
+import type { McpClientFactory } from './mcp';
+
+export interface InvokeModelArgs {
+  modelId: string;
+  options: LanguageModelV2CallOptions;
+}
+
+export interface InvokeModelResult {
+  content: Array<LanguageModelV2Content>;
+  finishReason: LanguageModelV2FinishReason;
+  usage: LanguageModelV2Usage;
+  providerMetadata?: SharedV2ProviderMetadata;
+  request?: { body?: unknown };
+  response?: LanguageModelV2ResponseMetadata & { headers?: SharedV2Headers; body?: unknown };
+  warnings: Array<LanguageModelV2CallWarning>;
+}
+
+export interface ListToolResult {
+  description?: string;
+  inputSchema: any;
+}
+
+export interface ListToolArgs {
+  clientArgs?: any;
+}
+
+export interface CallToolArgs {
+  clientArgs?: any;
+  name: string;
+  args: any;
+  options: ToolCallOptions;
+}
+
+/**
+ * Creates Temporal activities for AI model invocation using the provided AI SDK provider.
+ * These activities allow workflows to call AI models while maintaining Temporal's
+ * execution guarantees and replay safety.
+ *
+ * @param provider The AI SDK provider to use for model invocations
+ * @param mcpClientFactories A mapping of server names to functions to create mcp clients
+ * @returns An object containing the activity functions
+ *
+ * @experimental The AI SDK integration is an experimental feature; APIs may change without notice.
+ */
+export const createActivities = (provider: ProviderV2, mcpClientFactories?: [string, McpClientFactory][]): object => {
```

Comment:
nit: Please use the function syntax for top level definitions, rather than const+lambda form.

```suggestion
export function createActivities(provider: ProviderV2, mcpClientFactories?: [string, McpClientFactory][]): object {
```